src/egon/data/datasets/pypsaeursec/__init__.py

```diff
@@ -179,6 +179,90 @@ def read_network():
     return pypsa.Network(str(target_file))
 
 
+def clean_database():
+    """Remove all components abroad for eGon100RE of the database
+
+    Remove all components abroad and their associated time series of
+    the datase for the scenario 'eGon100RE'.
+
+    Parameters
+    ----------
+    None
+
+    Returns
+    -------
+    None
+
+    """
+    scn_name = "eGon100RE"
+
+    comp_one_port = ["load", "generator", "store", "storage"]
+
+    # delete existing components and associated timeseries
+    for comp in comp_one_port:
+        db.execute_sql(
+            f"""
+            DELETE FROM {"grid.egon_etrago_" + comp + "_timeseries"}
+            WHERE {comp + "_id"} IN (
+                SELECT {comp + "_id"} FROM {"grid.egon_etrago_" + comp}
+                WHERE bus IN (
+                    SELECT bus_id FROM grid.egon_etrago_bus
+                    WHERE country != 'DE'
+                    AND scn_name = '{scn_name}')
+                AND scn_name = '{scn_name}'
+            );
+
+            DELETE FROM {"grid.egon_etrago_" + comp}
+            WHERE bus IN (
+                SELECT bus_id FROM grid.egon_etrago_bus
+                WHERE country != 'DE'
+                AND scn_name = '{scn_name}')
+            AND scn_name = '{scn_name}';"""
+        )
+
+    comp_2_ports = [
+        "line",
+        "transformer",
+        "link",
+    ]
+
+    for comp, id in zip(comp_2_ports, ["line_id", "trafo_id", "link_id"]):
+        db.execute_sql(
+            f"""
+            DELETE FROM {"grid.egon_etrago_" + comp + "_timeseries"}
+            WHERE scn_name = '{scn_name}'
+            AND {id} IN (
+                SELECT {id} FROM {"grid.egon_etrago_" + comp}
+            WHERE "bus0" IN (
+            SELECT bus_id FROM grid.egon_etrago_bus
+                WHERE country != 'DE'
+                AND scn_name = '{scn_name}')
+            OR "bus1" IN (
+            SELECT bus_id FROM grid.egon_etrago_bus
+                WHERE country != 'DE'
+                AND scn_name = '{scn_name}')
+            );
+
+            DELETE FROM {"grid.egon_etrago_" + comp}
+            WHERE scn_name = '{scn_name}'
+            AND "bus0" IN (
+            SELECT bus_id FROM grid.egon_etrago_bus
+                WHERE country != 'DE'
+                AND scn_name = '{scn_name}')
+            OR "bus1" IN (
+            SELECT bus_id FROM grid.egon_etrago_bus
+                WHERE country != 'DE'
+                AND scn_name = '{scn_name}')
+            ;"""
```

Comment:
These statements delete not only the lines between foreign countries but also the ones between Germany and foreign countries. 
But pypsa-eur-sec is only used for the lines between foreign countries. The cross-border-lines from or to Germany are coming from osmTGmod. They shouldn't be deleted in this dataset. 
I think it would be enough to change `OR` to `AND` to fix this.

Reply:
Thanks for the remark, I 've changed it in https://github.com/openego/eGon-data/pull/848/commits/366b0baa8c9cd5922156abbcfb332ed2c647b5b8.